cmd/osbuild-worker/main.go

```diff
@@ -97,6 +98,10 @@ func main() {
 			OAuthURL         string `toml:"oauth_url"`
 			OfflineTokenPath string `toml:"offline_token"`
 		} `toml:"authentication"`
+		AWS *struct {
```

Comment:
I think this should be done like GCP and Azure, and simply take the path to the credentials file.

Reply:
+1, this struct is printed to stdout when a worker starts, thus we would leak credentials.

Reply:
Good for me.

Reply:
I'm wondering why we want to give the AWS credentials through the `toml` config file because, if we use the function to read credentials from files from the AWS library it reads them from the file pointed by `$AWS_SHARED_CREDENTIALS_FILE` or default to `$HOME/.aws/credentials`.

Reply:
So how would this work? I assume something like this?
1) Creds from a target would have the highest priority
2) Then, the library function would be called and it looks into `AWS_SHARED_CREDENTIALS_FILE` and `$HOME/.aws/credentials`.


Am I right?

Reply:
Exactly. Then the lib looks into `AWS_SHARED_CREDENTIALS_FILE` first, and if empty, looks in `$HOME/.aws/credentials`.

Reply:
Nice, I like that.

Reply:
btw does it work also with an attached IAM policy on a EC2 instance?

Reply:
For instance policies: if no creds are found using any of the methods described above, i believe we just need to pass `nil` as the creds object and the right thing should happen. I'm really not sure about that though, so something to check.

Reply:
By the way, only "problem" I could imagine with not specifying the file would be that it is inconsistent with how the others work. Maybe they can be made to work in similar ways too? Or maybe it does not matter. I would not block on it though.